crates/proof-of-sql-planner/src/expr.rs

```diff
@@ -0,0 +1,424 @@
+                )?),
+                Operator::Plus => Ok(DynProofExpr::try_new_add(
+                    left_proof_expr,
+                    right_proof_expr,
+                )?),
+                Operator::Minus => Ok(DynProofExpr::try_new_subtract(
+                    left_proof_expr,
+                    right_proof_expr,
+                )?),
+                Operator::Multiply => Ok(DynProofExpr::try_new_multiply(
+                    left_proof_expr,
+                    right_proof_expr,
+                )?),
+                Operator::And => Ok(DynProofExpr::try_new_and(
+                    left_proof_expr,
+                    right_proof_expr,
+                )?),
+                Operator::Or => Ok(DynProofExpr::try_new_or(left_proof_expr, right_proof_expr)?),
+                _ => Err(PlannerError::UnsupportedBinaryOperator { op: *op }),
+            }
+        }
+        Expr::Literal(val) => Ok(DynProofExpr::new_literal(scalar_value_to_literal_value(
+            val.clone(),
+        )?)),
+        Expr::Not(expr) => {
+            let proof_expr = expr_to_proof_expr(expr, schema)?;
+            Ok(DynProofExpr::try_new_not(proof_expr)?)
+        }
+        _ => Err(PlannerError::UnsupportedLogicalExpression { expr: expr.clone() }),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::df_util::*;
+    use arrow::datatypes::DataType;
+    use datafusion::{common::ScalarValue, logical_expr::expr::Placeholder};
+    use proof_of_sql::{base::database::ColumnType, sql::proof_exprs::utility::*};
+
+    #[allow(clippy::too_many_lines)]
+    #[test]
+    fn we_can_convert_expr_to_proof_expr() {
+        // Column
+        let expr = df_column(Some("namespace"), "table_name", "column");
+        let schema = df_schema("namespace.table_name", vec![("column", DataType::Int32)]);
+        assert_eq!(
+            expr_to_proof_expr(&expr, &schema).unwrap(),
+            column_expr(Some("namespace"), "table_name", "column", ColumnType::Int)
+        );
+
+        // BinaryExpr
+        // Eq
+        let expr = df_column(Some("namespace"), "table_name", "column1").eq(df_column(
+            Some("namespace"),
+            "table_name",
+            "column2",
+        ));
+        let schema = df_schema(
+            "namespace.table_name",
+            vec![("column1", DataType::Int16), ("column2", DataType::Int64)],
+        );
+        assert_eq!(
+            expr_to_proof_expr(&expr, &schema).unwrap(),
+            equal(
+                column_expr(
+                    Some("namespace"),
+                    "table_name",
+                    "column1",
+                    ColumnType::SmallInt
+                ),
+                column_expr(
+                    Some("namespace"),
+                    "table_name",
+                    "column2",
+                    ColumnType::BigInt
+                )
+            )
+        );
```

Comment:
NIT:
See my other comments. I kind of like this because it avoid "test utility" methods and it is very compact and readable.

The other benefit here of using `DynProofExpr::try_new_equals` is that it is obvious what the type is, where `equal(` is not clear without having to go find the code, or knowing what it is.

```suggestion
        assert_eq!(
            expr_to_proof_expr(&expr, &schema).unwrap(),
            DynProofExpr::try_new_equals(COLUMN1_SMALLINT(), COLUMN2_SMALLINT()).unwrap()
        );
```